components/clp-package-utils/clp_package_utils/scripts/compress.py

```diff
@@ -21,27 +26,177 @@
 logger = logging.getLogger(__file__)
 
 
+def _generate_logs_list(
+    container_logs_list_path: pathlib.Path,
+    parsed_args: argparse.Namespace,
+) -> None:
+    input_type = parsed_args.input_type
+
+    if InputType.FS == input_type:
+        host_logs_list_path = parsed_args.path_list
+        with open(container_logs_list_path, "w") as container_logs_list_file:
+            if host_logs_list_path is not None:
+                with open(host_logs_list_path, "r") as host_logs_list_file:
+                    for line in host_logs_list_file:
+                        stripped_path_str = line.rstrip()
+                        if "" == stripped_path_str:
+                            # Skip empty paths
+                            continue
+                        resolved_path = pathlib.Path(stripped_path_str).resolve()
+                        mounted_path = CONTAINER_INPUT_LOGS_ROOT_DIR / pathlib.Path(
+                            resolved_path
+                        ).relative_to(resolved_path.anchor)
+                        container_logs_list_file.write(f"{mounted_path}\n")
+
+            for path in parsed_args.paths:
+                resolved_path = pathlib.Path(path).resolve()
+                mounted_path = CONTAINER_INPUT_LOGS_ROOT_DIR / pathlib.Path(
+                    resolved_path
+                ).relative_to(resolved_path.anchor)
+                container_logs_list_file.write(f"{mounted_path}\n")
+
+    elif InputType.S3 == input_type:
+        with open(container_logs_list_path, "w") as container_logs_list_file:
+            container_logs_list_file.write(f"{parsed_args.url}\n")
+
+    else:
+        raise ValueError(f"Unsupported input type: {input_type}.")
+
+
+def _generate_compress_cmd(
+    parsed_args: argparse.Namespace, config_path: pathlib.Path, logs_list_path: pathlib.Path
+) -> List[str]:
+    input_type = parsed_args.input_type
+
+    # fmt: off
+    compress_cmd = [
+        "python3",
+        "-m", "clp_package_utils.scripts.native.compress",
+        "--config", str(config_path),
+        input_type,
+    ]
+    # fmt: on
+    if parsed_args.timestamp_key is not None:
+        compress_cmd.append("--timestamp-key")
+        compress_cmd.append(parsed_args.timestamp_key)
+    if parsed_args.tags is not None:
+        compress_cmd.append("--tags")
+        compress_cmd.append(parsed_args.tags)
+    if parsed_args.no_progress_reporting is True:
+        compress_cmd.append("--no-progress-reporting")
+
+    if InputType.FS == input_type:
+        pass
+    elif InputType.S3 == input_type:
+        aws_access_key_id = parsed_args.aws_access_key_id
+        aws_secret_access_key = parsed_args.aws_secret_access_key
+        if parsed_args.aws_credentials_file:
+            default_credentials_user = "default"
```

Comment:
a credential file can contain multiple credentials, such as the follow (copied from https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-files.html)

Should we use "default" for now, or let user pass in a user? Technically they can always create a new credentials file with only [default] section 

```
[default]
aws_access_key_id=ASIAIOSFODNN7EXAMPLE
aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
aws_session_token = IQoJb3JpZ2luX2IQoJb3JpZ2luX2IQoJb3JpZ2luX2IQoJb3JpZ2luX2IQoJb3JpZVERYLONGSTRINGEXAMPLE

[user1]
aws_access_key_id=ASIAI44QH8DHBEXAMPLE
aws_secret_access_key=je7MtGbClwBF/2Zp9Utk/h3yCo8nvbEXAMPLEKEY
aws_session_token = fcZib3JpZ2luX2IQoJb3JpZ2luX2IQoJb3JpZ2luX2IQoJb3JpZ2luX2IQoJb3JpZVERYLONGSTRINGEXAMPLE

```